sim/pokemon.ts

```diff
@@ -19,29 +19,100 @@ interface MoveSlot {
 }
 
 export class Pokemon {
-	side: Side;
-	battle: Battle;
-	set: PokemonSet;
-	getHealth: (side: Side) => string;
-	getDetails: (side: Side) => string;
+	readonly side: Side;
+	readonly battle: Battle;
+
+	/** "pre-bound" functions for nicer syntax allows them to be passed directly to Battle#add. */
+	readonly getHealth: (side: Side) => string;
+	readonly getDetails: (side: Side) => string;
+
+	readonly set: PokemonSet;
+	readonly name: string;
+	readonly fullname: string;
+	readonly id: string; // shouldn't really be used anywhere
+	readonly species: string;
+	readonly speciesid: string;
+	readonly level: number;
+	readonly gender: GenderName;
+	readonly happiness: number;
+	readonly pokeball: string;
+
+	/** Transform keeps the original pre-transformed Hidden Power in Gen 2-4. */
+	readonly baseHpType: string;
+	readonly baseHpPower: number;
+
+	readonly baseMoveSlots: MoveSlot[];
+	moveSlots: MoveSlot[];
+
+	hpType: string;
+	hpPower: number;
+
+	position: number;
+	details: string;
+
 	baseTemplate: Template;
-	species: string;
-	name: string;
-	speciesid: string;
 	template: Template;
-	moveSlots: MoveSlot[];
-	baseMoveSlots: MoveSlot[];
+	speciesData: AnyObject;
+
+	status: string;
+	statusData: AnyObject;
+	volatiles: AnyObject;
+	showCure: boolean;
+
+	/**
+	 * These are the basic stats that appear on the in-game stats screen:
+	 * calculated purely from the species base stats, level, IVs, EVs,
+	 * and Nature, before modifications from item, ability, etc.
+	 *
+	 * `stats` are reset to `baseStats` on switch-out.
+	 */
 	baseStats: StatsTable;
```

Comment:
```suggestion
	readonly baseStats: StatsTable;
```

You mark all the other `base` properties as read-only; this should be no different.

Reply:
Wait, nvm, this is modified by Mega Evolution and other effects that change things permanently.